packages/open-next/src/build.ts

```diff
@@ -641,151 +688,83 @@ function createCacheAssets(monorepoRoot: string, disableDynamoDBCache = false) {
-      }),
-    ];
-  }
+    // Copy open-next.config.mjs
+    copyOpenNextConfig(options.tempDir, outputPath);
 
-  if (streaming) {
-    const streamingPlugin = openNextPlugin({
-      name: "opennext-streaming",
-      target: /plugins\/lambdaHandler\.js/g,
-      replacements: ["./streaming.replacement.js"],
+    // Bundle middleware
+    await buildEdgeBundle({
+      entrypoint: path.join(__dirname, "adapters", "middleware.js"),
+      outfile: path.join(outputPath, "handler.mjs"),
+      ...commonMiddlewareOptions,
+      overrides: config.middleware?.override,
+      defaultConverter: "aws-cloudfront",
+    });
+  } else {
+    await buildEdgeBundle({
+      entrypoint: path.join(__dirname, "core", "edgeFunctionHandler.js"),
+      outfile: path.join(outputDir, ".build", "middleware.mjs"),
+      ...commonMiddlewareOptions,
     });
-    if (plugins) {
-      plugins.push(streamingPlugin);
-    } else {
-      plugins = [streamingPlugin];
-    }
-  }
-
-  if (plugins && plugins.length > 0) {
-    logger.debug(
-      `Applying plugins:: [${plugins
-        .map(({ name }) => name)
-        .join(",")}] for Next version: ${options.nextVersion}`,
-    );
-  }
-  await esbuildAsync({
-    entryPoints: [path.join(__dirname, "adapters", "server-adapter.js")],
-    external: ["next"],
-    outfile: path.join(outputPath, packagePath, "index.mjs"),
-    banner: {
-      js: [
-        `globalThis.monorepoPackagePath = "${packagePath}";`,
-        "import { createRequire as topLevelCreateRequire } from 'module';",
-        "const require = topLevelCreateRequire(import.meta.url);",
-        "import bannerUrl from 'url';",
-        "const __dirname = bannerUrl.fileURLToPath(new URL('.', import.meta.url));",
-      ].join(""),
-    },
-    plugins,
-  });
-
-  if (isMonorepo) {
-    addMonorepoEntrypoint(outputPath, packagePath);
   }
-  addPublicFilesList(outputPath, packagePath);
-  injectMiddlewareGeolocation(outputPath, packagePath);
-  removeCachedPages(outputPath, packagePath);
-  addCacheHandler(outputPath, options.dangerous);
-}
-
-function addMonorepoEntrypoint(outputPath: string, packagePath: string) {
-  // Note: in the monorepo case, the handler file is output to
-  //       `.next/standalone/package/path/index.mjs`, but we want
-  //       the Lambda function to be able to find the handler at
-  //       the root of the bundle. We will create a dummy `index.mjs`
-  //       that re-exports the real handler.
-
-  // Always use posix path for import path
-  const packagePosixPath = packagePath.split(path.sep).join(path.posix.sep);
-  fs.writeFileSync(
-    path.join(outputPath, "index.mjs"),
-    [
-      `export const handler = async (event, context) => {`,
-      ` const fn = await import("./${packagePosixPath}/index.mjs");`,
-      ` return fn.handler(event, context);`,
-      `};`,
-    ].join(""),
-  );
 }
 
+//TODO: Why do we need this? People have access to the headers in the middleware
```

Comment:
I forgot to remove this one, this is handled differently now https://github.com/sst/open-next/pull/402/commits/9f7e1084b667eaf533adc941e4c0a950a70977fd
It will not work anymore for next < 13.4.13 since it is handled in the middleware directly, but they still have access to the cloudfront headers